File: methodoverride.cpp
#include<iostream>
using namespace std;

class Base {
public:
    void add(int a, int b) {
        cout << "Base Class: " << a + b << endl;
    }
    
    void add(double a, double b) {
        cout << "Base Class: " << a + b << endl;
    }
};

class Override : public Base {
public:
    void add(int a, int b) {
        cout << "Override Class: " << a + b << endl;
    }
    
    void add(double a, double b) {
        cout << "Override Class: " << a + b << endl;
    }
};

int main() {
    Base obj;
    obj.add(2, 4);
    obj.add(7.4, 67.5);

    Override obj2;
    obj2.add(2, 4);
    obj2.add(7.4, 67.5);
    
    return 0;
}

File: methodoverride.java
class Base {
    public void add(int a, int b) {
        System.out.println("Base Class: " + (a + b));
    }
    
    public void add(double a, double b) {
        System.out.println("Base Class: " + (a + b));
    }
}

class Override extends Base {
    public void add(int a, int b) {
        System.out.println("Override Class: " + (a + b));
    }
    
    public void add(double a, double b) {
        System.out.println("Override Class: " + (a + b));
    }
}

public class Main {
    public static void main(String[] args) {
        Base obj = new Base();
        obj.add(2, 4);
        obj.add(7.4, 67.5);

        Override obj2 = new Override();
        obj2.add(2, 4);
        obj2.add(7.4, 67.5);
    }
}
